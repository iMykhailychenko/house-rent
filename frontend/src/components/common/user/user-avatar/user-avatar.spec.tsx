import { render } from '@testing-library/react';

import UserAvatar from './user-avatar';

describe('UserAvatar component', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<UserAvatar firstName="firstName" lastName="lastName" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('without img', () => {
        const { getByText, queryByTitle } = render(<UserAvatar firstName="firstName" lastName="lastName" />);
        expect(getByText('fl')).toBeInTheDocument();
        expect(queryByTitle('User avatar')).toBeFalsy();
    });

    it('with img', () => {
        const { queryByText, getByTitle } = render(<UserAvatar src="/test" firstName="firstName" lastName="lastName" />);
        expect(queryByText('fl')).toBeFalsy();
        expect(getByTitle('User avatar')).toBeInTheDocument();
    });
});
