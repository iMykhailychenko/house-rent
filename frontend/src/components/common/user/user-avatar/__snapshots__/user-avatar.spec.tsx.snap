// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="img name"
    style="height: 4rem; width: 4rem;"
  >
    fl
  </div>
</DocumentFragment>
`;
